### PR TITLE
Adopt Ruff linter and formatter 

### PR DESCRIPTION
**Pull request by juanitorduz**
Closes https://github.com/pymc-labs/CausalPy/issues/267

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ----
 
 ![Build](https://github.com/pymc-labs/CausalPy/workflows/ci/badge.svg)
-[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
+[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v1.json)]([https://codecov.io/gh/pymc-labs/pymc-marketing](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v1.json))
```
Comment (twiecki) [inline README.md:8]: Wrong URL
```suggestion
[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v1.json)]([https://codecov.io/gh/pymc-labs/CausalPy](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/charliermarsh/ruff/main/assets/badge/v1.json))
```

**Review by drbenvincent — APPROVED** [review]
Looks good. Great that we can get rid of `setup.cfg`, the root dir is now about as minimal as we can get it.

**Comment by drbenvincent** [comment]
Nice. Have also swapped out my Black VS Code extension for the Ruff extension

File: README.md (hunk 1 above)
Comment (juanitorduz) [reply README.md]: Ups! Thanks for fixing it!

**Pull request merged**